ios/CodePush/CodePushPackage.m

```diff
@@ -481,28 +537,33 @@ + (void)installPackage:(NSDictionary *)updatePackage
     }
     
     [info setValue:packageHash forKey:@"currentPackage"];
-    
+
+    //TODO: This method should return a BOOL indication of success.
     [self updateCurrentPackageInfo:info
-                             error:error];
+                             error:&localError];
+
+    if (localError && error) {
+        *error = localError;
+    }
 }
 
 + (void)rollbackPackage
 {
     NSError *error;
     NSMutableDictionary *info = [self getCurrentPackageInfo:&error];
-    if (error) {
+    if (!info) {
         return;
     }
     
     NSString *currentPackageFolderPath = [self getCurrentPackageFolderPath:&error];
-    if (error) {
+    if (!currentPackageFolderPath) {
         return;
     }
     
     NSError *deleteError;
-    [[NSFileManager defaultManager] removeItemAtPath:currentPackageFolderPath
+    BOOL result = [[NSFileManager defaultManager] removeItemAtPath:currentPackageFolderPath
                                                error:&deleteError];
-    if (deleteError) {
+    if (!result) {
```

Comment:
This probably isn't part of your change scope, but I think we should print the error too.